src/cloudai/_core/test.py

```diff
@@ -122,30 +111,6 @@ def __repr__(self) -> str:
             f"extra_cmd_args={self.extra_cmd_args}, "
             f"section_name={self.section_name}, "
             f"dependencies={self.dependencies}, iterations={self.iterations}, "
-            f"nodes={self.nodes})"
-        )
-
-    def gen_exec_command(self, output_path: str) -> str:
-        """
-        Generate the command to run this specific test.
-
-        Args:
-            output_path (str): Path to the output directory.
-
-        Returns:
-            str: The command string.
-        """
-        if self.time_limit is not None:
-            self.cmd_args["time_limit"] = self.time_limit
-
-        return self.test_template.gen_exec_command(
-            self.env_vars,
-            self.cmd_args,
-            self.extra_env_vars,
-            self.extra_cmd_args,
-            output_path,
-            self.num_nodes,
-            self.nodes,
         )
 
     def get_job_id(self, stdout: str, stderr: str) -> Optional[int]:
```

Comment:
If the TestRun class represents a test that is running, then we need to move these methods (get_job_id and get_job_status) to the TestRun class as well.

Reply:
Yes, that is the plan. And believe there are more fields like this.

Reply:
Actually, these to methods could be part of Runner class, seems the fit there better. But looks like `iterations` could be moved to `TestRun` as part of this PR, what do you think?

Reply:
It makes sense.